### PR TITLE
Fix/comment expand error

### PR DESCRIPTION
**Pull request by pitb2022**

### CODE REVIEW

File: src/components/Expandable/index.tsx [hunk 1]
```diff
@@ -73,45 +127,47 @@ export const Expandable: React.FC<ExpandableProps> = ({
...
+    handleOverflowCheck()
   }, [content])
 
   useEffect(() => {
-    if (firstRender) {
-      setFirstRender(false)
+    if (isSafariVersionLessThan17 && isRichShow) {
+      reset()
+      setIsRichShow(false)
+      setTimeout(() => {
```
Comment (robertu7) [inline src/components/Expandable/index.tsx:159]: should avoid using `setTimeout`, can add another `useEffect` hook or merge it into L148 instead:

```tsx
useEffect(() => {
    handleOverflowCheck()
}, [...])
```

or 

```tsx
useIsomorphicLayoutEffect(() => {
  reset()
  handleOverflowCheck()
}, [content, isRichShow])
```

**Review by robertu7 — CHANGES_REQUESTED** [review]

File: src/common/utils/detect.ts [hunk 2]
```diff
@@ -13,3 +13,15 @@ export const isMobile = () => {
     userAgent
   )
 }
+
+export const checkIsSafariVersionLessThan17 = () => {
+  const userAgent = navigator.userAgent
+  const safariMatch = userAgent.match(/Version\/(\d+\.\d+)/)
```
[the remark this answers is not in the document]
Comment (pitb2022) [reply src/common/utils/detect.ts:19]: This is not applicable to Apple M series chips?

[the remark this answers is not in the document]
Comment (robertu7) [reply src/common/utils/detect.ts]: Here is mine: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.5 Safari/605.1.15`

File: src/components/Expandable/index.tsx [hunk 3]
```diff
@@ -45,23 +73,33 @@ export const Expandable: React.FC<ExpandableProps> = ({
...
+  isRichShow: _isRichShow = false,
+  isComment = false,
   collapseable = true,
   bgColor = 'white',
 }) => {
-  const [expandable, setExpandable] = useState(false)
-  const [firstRender, setFirstRender] = useState(true)
-  const [expand, setExpand] = useState(true)
+  const [isOverFlowing, setIsOverFlowing] = useState(false)
+  const [isExpand, setIsExpand] = useState(true)
```
Comment (robertu7) [inline src/components/Expandable/index.tsx:82]: Expand => Expanded

**Review by robertu7 — APPROVED** [review]

**Pull request merged**